src/fs/patch.ts

```diff
@@ -1,329 +1,76 @@
-          function (error: Error, bytesRead: number, result: Buffer) {
-            if (error) {
-              return originalFsMethods.close(fd, function () {
-                callback(error, null)
-              })
-            }
-            originalFsMethods.close(fd, function (err: Error) {
-              if (err) {
-                return callback(err, result)
-              }
-              callback(err, encoding ? result.toString(encoding) : result)
-            })
-          }
-        )
-      })
-    },
-    createReadStream: function createReadStream(filepath: any, options: any) {
-      setupManifest()
-      const entry = manifest[getKey(filepath)]
-      if (!entry) {
-        return originalFsMethods.createReadStream.apply(fs, arguments)
-      }
-      const [offset, length] = entry
-      const resourceOffset = resourceStart + offset
-      const opts = fileOpts(options)
-
-      return originalFsMethods.createReadStream(
-        blobPath,
-        Object.assign({}, opts, {
-          start: resourceOffset,
-          end: resourceOffset + length - 1,
-        })
-      )
-    },
-    readFileSync: function readFileSync(filepath: any, options: any) {
-      setupManifest()
-      const entry = manifest[getKey(filepath)]
-      if (!entry) {
-        return originalFsMethods.readFileSync.apply(fs, arguments)
-      }
-      const [offset, length] = entry
-      const resourceOffset = resourceStart + offset
-      const encoding = fileOpts(options).encoding
-      const fd = originalFsMethods.openSync(process.execPath, 'r')
-      const result = Buffer.alloc(length)
-      originalFsMethods.readSync(fd, result, 0, length, resourceOffset)
-      originalFsMethods.closeSync(fd)
-      return encoding ? result.toString(encoding) : result
-    },
-    statSync: function statSync(filepath: string | Buffer, options: any) {
-      const stat = ownStat(filepath, options)
-      if (stat) {
-        return stat
-      }
-      return originalFsMethods.statSync.apply(fs, arguments)
-    },
-    stat: getStat('stat'),
-    lstat: getStat('lstat'),
-    lstatSync: function statSync(filepath: string | Buffer, options: any) {
-      const stat = ownStat(filepath, options)
-      if (stat) {
-        return stat
-      }
-      return originalFsMethods.lstatSync.apply(fs, arguments)
-    },
-  }
-  if (typeof fs.exists === 'function') {
-    nfs.exists = function (filepath: string, cb: Function) {
-      cb = cb || noop
-      const exists = nfs.existsSync(filepath)
-      process.nextTick(() => cb(exists))
-    }
-  }
-
-  const patches = (process as any).nexe.patches || {}
-  delete (process as any).nexe
   patches.internalModuleReadFile = function (this: any, original: any, ...args: any[]) {
-    setupManifest()
-    const filepath = getKey(args[0])
-    if (manifest[filepath]) {
-      log('read     (hit)              ' + filepath)
-      return nfs.readFileSync(filepath, 'utf-8')
+    log(`internalModuleReadFile ${args[0]}`)
```

Comment:
I don't know if this is sufficient. I think we still need getKey semantics (to map to a `/snapshot` based path.
then IFF the file doesn't exist in the zipfs, fallback to the `original.call..`

Reply:
I think we could include the manifest at build time to avoid seeking for it at startup.

Reply:
> I don't know if this is sufficient. I think we still need getKey semantics to map to a /snapshot based path.
then IFF the file doesn't exist in the zipfs, fallback to the original.call..

Was looking into this, and it pointed out an incompatibility that I'd missed...

This branch/PR (which I was thinking I'd close and replace with https://github.com/nexe/nexe/pull/1048 - although I'm happy to pull your feedback from here across to that branch if reviewing here is easier :-) ) was resolving files relative to `process.cwd()` whereas master currently does it relative to `path.dirname(process.argv[0])` (essentially)... After solving that, this is then all handled by the SnapshotZipFS implementation because it will lookup the file in the zip and return it IFF it's there, otherwise it'll go to the real fs - so the there's no need for the special casing in here, and it can just be passed through :-) (or, at least, afaiu :-) )

(This also needed a tweak to `realpathSync` in SnapshotZipFS to ensure that the module path doesn't come back as the `/snapshot` path :-) )

Reply:
Oh, I didn't see #1048. Can we close one or the other?